### PR TITLE
feat: update flask to version 3.0

### PR DESCRIPTION
**Pull request by joaoandre-avaiga**

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -6,22 +6,20 @@ name = "pypi"
...
-Flask-RESTful = "==0.3.9"
-flask-marshmallow = "==0.14"
+Flask-Migrate = ">=3.0"
+Flask-RESTful = ">=0.3.9"
+flask-marshmallow = ">=0.14"
 markupsafe = "==2.1.1"
 marshmallow-sqlalchemy = "==0.25"
 passlib = "==1.7.4"
 taipy-core = {ref = "develop", git = "https://github.com/avaiga/taipy-core.git"}
+flask = "==3.0"
```
Comment (trgiangdo) [inline Pipfile]:
```suggestion
flask = "==3.0.0"
```
Minor thing. Just to completely match with taipy-gui

**Review by trgiangdo — COMMENTED** [review]
Minor comment.

We also need to update the setup.py file to match taipy-gui
```
"flask>=3.0.0,<3.1",
```

File: Pipfile [hunk 2]
```diff
@@ -6,22 +6,20 @@ name = "pypi"
 [packages]
 apispec = {extras = ["yaml"], version = "==5.0"}
 apispec-webframeworks = "==0.5.2"
-flask = "==2.3.3"
-Flask-JWT-Extended = "==4.3"
```
Comment (jrobinAV) [inline Pipfile:10]: It makes me think we should test rest with authentication as well. Do we have an integration test for that ?

**Review by jrobinAV — DISMISSED** [review]

File: Pipfile (hunk 2 above)
Comment (joaoandre-avaiga) [reply Pipfile:10]: Good question. I think not.

Comment (jrobinAV) [reply Pipfile:10]: Let's keep that in mind

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**